extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js

```diff
@@ -0,0 +1,506 @@
+    if (nextProps.responsiveView != this.props.responsiveView) {
+      this.responsiveUIMultiple = this.getResponsiveUIMultiple(nextProps.responsiveView);
+    }
+  },
+
+  componentWillUnmount: function () {
+    this.props.controller.cancelTimer();
+    this.closePopovers();
+    if (Utils.isAndroid()){
+      this.props.controller.hideVolumeSliderBar();
+    }
+    window.removeEventListener('orientationchange', this.closePopovers);
+  },
+
+  getResponsiveUIMultiple: function(responsiveView){
+    var multiplier = this.props.skinConfig.responsive.breakpoints[responsiveView].multiplier;
+    return multiplier;
+  },
+
+  handleControlBarMouseUp: function(evt) {
+    if (evt.type == 'touchend' || !this.isMobile){
+      evt.stopPropagation(); // W3C
+      evt.cancelBubble = true; // IE
+      this.props.controller.state.accessibilityControlsEnabled = true;
+      this.props.controller.startHideControlBarTimer();
+    }
+  },
+
+  handleFullscreenClick: function(evt) {
+    // On mobile, we get a following click event that fires after the Video
+    // has gone full screen, clicking on a different UI element. So we prevent
+    // the following click.
+    evt.stopPropagation();
+    evt.cancelBubble = true;
+    evt.preventDefault();
+    this.props.controller.toggleFullscreen();
+  },
+
+  handleLiveClick: function(evt) {
+    evt.stopPropagation();
+    evt.cancelBubble = true;
+    evt.preventDefault();
+    this.props.controller.onLiveClick();
+    this.props.controller.seek(this.props.duration);
+  },
+
+  handleVolumeIconClick: function(evt) {
+    if (this.isMobile){
+      this.props.controller.startHideControlBarTimer();
+      evt.stopPropagation(); // W3C
+      evt.cancelBubble = true; // IE
+      var canShowVolumeSliderBar = !this.props.skinConfig.controlBar
+        || !this.props.skinConfig.controlBar.volumeControl
+        || this.props.skinConfig.controlBar.volumeControl.sliderVisible !== false;
+      if (canShowVolumeSliderBar && !this.props.controller.state.volumeState.volumeSliderVisible){
+        this.props.controller.showVolumeSliderBar();
+      }
+      else {
+        this.props.controller.handleMuteClick();
+      }
+    }
+    else{
+      this.props.controller.handleMuteClick();
+    }
+  },
+
+  handlePlayClick: function() {
+    this.props.controller.togglePlayPause();
+  },
+
+  handleShareClick: function() {
+    this.props.controller.toggleShareScreen();
+  },
+
+  handleQualityClick: function() {
+    if(this.props.responsiveView == this.props.skinConfig.responsive.breakpoints.xs.id) {
+      this.props.controller.toggleScreen(CONSTANTS.SCREEN.VIDEO_QUALITY_SCREEN);
+    } else {
+      this.toggleQualityPopover();
+      this.closeCaptionPopover();
+    }
+  },
+
+  toggleQualityPopover: function() {
+    this.props.controller.toggleVideoQualityPopOver();
+  },
+
+  closeQualityPopover: function() {
+    if(this.props.controller.state.videoQualityOptions.showVideoQualityPopover == true) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace "==" with "===". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "true" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1125)